modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -213,7 +221,10 @@ GrainTracker::finalize()
       {
         std::set<dof_id_type>::const_iterator elem_it_end = grain_it->second->entities_ptr->end();
         for (std::set<dof_id_type>::const_iterator elem_it = grain_it->second->entities_ptr->begin(); elem_it != elem_it_end; ++elem_it)
-          _elemental_data[*elem_it].push_back(std::make_pair(grain_it->first, grain_it->second->variable_idx));
+        {
+          mooseAssert(!_ebsd_reader || _unique_grain_to_ebsd_num.find(grain_it->first) != _unique_grain_to_ebsd_num.end(), "Bad mapping in unique_grain_to_ebsd_num");
+          _elemental_data[*elem_it].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_it->first] : grain_it->first, grain_it->second->variable_idx));
```

Comment:
Good question, if it does then something is very wrong. That map is a
lookup of the grain id to the ebsd number. I figured I'd be lazy and just
assert the lookup.
On Wed, Sep 9, 2015 at 5:00 PM John W. Peterson <notifications@github.com>
wrote:

> In modules/phase_field/src/postprocessors/GrainTracker.C
> <https://github.com/idaholab/moose/pull/5675#discussion_r39107734>:
>
> > @@ -213,7 +221,10 @@ GrainTracker::finalize()
> >        {
> >          std::set<dof_id_type>::const_iterator elem_it_end = grain_it->second->entities_ptr->end();
> >          for (std::set<dof_id_type>::const_iterator elem_it = grain_it->second->entities_ptr->begin(); elem_it != elem_it_end; ++elem_it)
> > -          _elemental_data[*elem_it].push_back(std::make_pair(grain_it->first, grain_it->second->variable_idx));
> > +        {
> > +          mooseAssert(!_ebsd_reader || _unique_grain_to_ebsd_num.find(grain_it->first) != _unique_grain_to_ebsd_num.end(), "Bad mapping in unique_grain_to_ebsd_num");
> > +          _elemental_data[*elem_it].push_back(std::make_pair(_ebsd_reader ? _unique_grain_to_ebsd_num[grain_it->first] : grain_it->first, grain_it->second->variable_idx));
>
> Are you sure this doesn't do accidental insertion?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5675/files#r39107734>.
>